### PR TITLE
[REVIEW] Update Java slf4j version to match Spark 3.0 [skip ci]

### PR DESCRIPTION
**Pull request by tgravescs**
Update the java slf4j version to 1.7.30 to match Spark 3.0 version number so we don't have any conflicts with the versioning.

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: java/pom.xml [hunk 1]
```diff
@@ -105,7 +105,7 @@
         <dependency>
             <groupId>org.slf4j</groupId>
             <artifactId>slf4j-api</artifactId>
-            <version>1.7.9</version>
+            <version>1.7.30</version>
```
Comment (jlowe) [inline java/pom.xml:108]: Nit: we should make this a property as was done for `junit.version` to make keeping the various slf4j artifacts needed easier to keep in sync.

**Comment by tgravescs** [comment]
need to fix spacing

**Pull request merged**